cmds/upspin_sos/service.go

```diff
@@ -78,9 +78,23 @@ func (us UpspinService) setFileData(path string) error {
 }
 
 func (us UpspinService) setKeys(path string) error {
-	// execute as user. This command generates files with elevated permissions otherwise
-	keygen := exec.Command("upspin", "keygen", fmt.Sprintf("-secretseed=%v", us.Seed), path)
-	err := keygen.Run()
+	// check if keys are set already
+	f, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+	_, err = f.Readdir(1)
```

Comment:
I found the ordering here a bit confusing for a bit. 
Consider doing this
args := whatever
if _, err := f.Readdir; err == nil {
args = append(args, "-rotate")
}